### PR TITLE
Decoder encoder

### PR DESCRIPTION
**Pull request by Voytech-0**

### CODE REVIEW

File: src/feat_engineering.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import pandas as pd
 import matplotlib.pyplot as plt
+from tqdm import tqdm
 
 
-def eliminate_long_sentences(data: pd.DataFrame, max_length: int = 400) -> pd.DataFrame:
+def eliminate_long_sentences(data: pd.DataFrame, max_length: int = 80) -> pd.DataFrame:
```
Comment (LauraMQuiros) [inline src/feat_engineering.py:6]: why this change in default value? Have you also deleted small sentences?

File: src/seq2seq.ipynb [hunk 2]
```diff
@@ -0,0 +1,686 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 56,
```
Comment (LauraMQuiros) [inline src/seq2seq.ipynb:5]: Error appears on github with  the following: Invalid Notebook: 'execution_count' is a required property, i'm unsure why

File: src/seq2seq.ipynb [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+    "        self.n_words = 2  # Count SOS and EOS\n",
+    "\n",
+    "    def addSentence(self, sentence):\n",
+    "        for word in sentence.split(' '):\n",
+    "            self.addWord(word)\n",
+    "\n",
+    "    def addWord(self, word):\n",
+    "        if word not in self.word2index:\n",
+    "            self.word2index[word] = self.n_words\n",
+    "            self.word2count[word] = 1\n",
```
Comment (LauraMQuiros) [inline src/seq2seq.ipynb:99]: then wouldn't each unique word have value 1? since this is an assignment, not an increment

File: src/seq2seq.ipynb [hunk 4]
```diff
@@ -0,0 +1,686 @@
...
+  },
+  {
+   "cell_type": "code",
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Reading lines...\n",
+      "('In my opinion the European Union should have the right to be informed of the minimum levels of social benefits in the individual Member States ', 'Teniendo esto en mente el Consejo Europeo sugirió un límite del 1,5 para que el déficit de transposición de las directivas se aplicara a todos los Estados miembros ')\n"
```
Comment (LauraMQuiros) [inline src/seq2seq.ipynb:163]: Should we be concerned that this is not a correct translation?

**Review by LauraMQuiros — COMMENTED** [review]
I'm  missing more comments on the Lang class and further functions (maybe consider javadoc?). Also would love if we had a README with more description on how is the flow of data in the model (like a description of where the input goes to, how is the model architecture, what is the output). Also if you need to make more preprocessing before the data is fed to the model, maybe we should consider moving these functions to the preprocessing file.

File: src/seq2seq.ipynb (hunk 3 above)
Comment (Voytech-0) [reply src/seq2seq.ipynb:99]: it is an assignment for words that are not yet in the dictionary. We increment for words that are already there

File: src/seq2seq.ipynb (hunk 4 above)
Comment (Voytech-0) [reply src/seq2seq.ipynb:163]: That appears in en_es_translation generated by your preprocessing. Could you check whether the translations roughly match the meaning? I do not know whether we randomly assigned the pair or it is an error within europarl-v7 document

File: src/feat_engineering.py (hunk 1 above)
Comment (LauraMQuiros) [reply src/feat_engineering.py:6]: I don't recall that. Is it justified, given hat most of the sentences are under 80, with the mean in both datasets are under 50? Like I don't think we can say it's outliers

File: src/seq2seq.ipynb (hunk 3 above)
Comment (LauraMQuiros) [reply src/seq2seq.ipynb:99]: oooh alrighty

File: src/feat_engineering.py (hunk 1 above)
Comment (Voytech-0) [reply src/feat_engineering.py:6]: We discussed it during the lab (I think two weeks ago). You said that most sentences are under 80. We can increase it, but that will also increase the training time. Since you are responsible for training it, I would say it is your call to increase it if you think it is necessary

File: src/seq2seq.ipynb (hunk 4 above)
Comment (LauraMQuiros) [reply src/seq2seq.ipynb:163]: There seems to be, indeed, an error in the preprocessing. The early translations are correct, but probably due to the fact that we are not eliminating sentences in a pair-wise manner, this causes an offset. I  will fix this before our next meeting and make a PR to change the preprocessing file. This shouldn't affect your PR.

File: src/feat_engineering.py (hunk 1 above)
Comment (LauraMQuiros) [reply src/feat_engineering.py:6]: Lmao I must've said it without looking at the boxplot. I think we shouldn't eliminate any short sentences, but also lack of computational resources may be a valid reason why we can eliminate short sentences. At the same time, our model may struggle more without the short sentences, since long sentences have more connectives and manners of speech. Should we ask our TA for advice?

Comment (LauraMQuiros) [reply src/feat_engineering.py:6]: maybe we should make an issue and assign it to him? idk if that's ok

**Review by LauraMQuiros — APPROVED** [review]
Looks good. The notebook is still not available, but i guess we can ignore that
